### PR TITLE
[PRD-4160] Xul based DrillDown dialogs in Report Designer cannot be internationalized

### PR DESCRIPTION
**Pull request by mr9d**
1. Swing analogs for following dialogs are implemented:
  - self-drilldown.xul
  - generic-url-drilldown.xul
  - pentaho-sugar-drilldown.xul
2. Following settings were added to 'messages' resource bundle for i18n purpose:
  - DrillDownDialog.PathInput.Label=Path:
  - DrillDownDialog.ServerUrlCheckBox.Text=Include server URL in path
  - DrillDownDialog.ServerUrlInput.Label=Server URL:
  - DrillDownDialog.LoginButton.Text=Login
  - DrillDownDialog.BrowseButton.Text=Browse
  - DrillDownParameterTable.ParameterName=Name
  - DrillDownParameterTable.ParameterValue=Value
  - DrillDownParameterTable.Title=Parameters:

Previous PR for this issue is https://github.com/pentaho/pentaho-reporting/pull/741

### CODE REVIEW

**Comment by mr9d** [comment]
@tmorgner please review.

File: designer/report-designer/src/org/pentaho/reporting/designer/core/editor/drilldown/swing/SwingGenericUrlDrillDownUiProfile.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+ *
+ * @author Aleksandr Kozlov
+ */
+public class SwingGenericUrlDrillDownUiProfile implements DrillDownUiProfile {
+
+  /** Drill down meta data. */
+  private DrillDownProfile drillDownProfile;
+
+  /** The name of profile for SwingGenericUrlDrillDownUi. */
+  public static final String[] NAMES = { "generic-url", "local-url" };
```
Comment (tmorgner) [inline designer/report-designer/src/org/pentaho/reporting/designer/core/editor/drilldown/swing/SwingGenericUrlDrillDownUiProfile.java]: Public static arrays are always a bad thing. Anyone (even from scripts running in a report or kettle-step) can do bad things to the contents of the array. 

Can you rewrite this (and the other occurences) to use a read-only list instead? 
Collections.unmodifiableList(Arrays.asList(....)) should do the trick and no one can ever change the contents of the list.

**Comment by tmorgner** [comment]
Apart from the public static array, this looks great.

File: designer/report-designer/src/org/pentaho/reporting/designer/core/editor/drilldown/swing/SwingGenericUrlDrillDownUiProfile.java (hunk 1 above)
Comment (mr9d) [inline designer/report-designer/src/org/pentaho/reporting/designer/core/editor/drilldown/swing/SwingGenericUrlDrillDownUiProfile.java]: Fixed.

**Comment by mr9d** [comment]
Fixed. I've removed all public static final arrays.

**Comment by tmorgner** [comment]
And I merged. We are even! :)

**Pull request merged**